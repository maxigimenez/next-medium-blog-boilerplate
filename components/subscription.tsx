import React, { useState } from 'react';
import config from '@/config';

interface AlertMeta {
  class: string;
  message: string;
}

export const Subscription = () => {
  const [email, setEmail] = useState<string>('');
  const [blocked, setBlocked] = useState<boolean>(false);
  const [alert, setAlert] = useState<AlertMeta>();

  const subscribe = async () => {
    setBlocked(true);
    setAlert(null);
    try {
      const data = await fetch(config.subscription.url, {
        method: 'POST',
        body: JSON.stringify({ email, date: new Date() }),
        headers: {
          'Content-Type': 'application/json',
        },
      });
      const json = await data.json();
      if (json) {
        setAlert({
          class: 'success',
          message: config.subscription.success,
        });
      }
    } catch (e) {
      setAlert({
        class: 'danger',
        message: config.subscription.error,
      });
      setBlocked(false);
    }
  };

  return (
    <div className="border p-5 bg-lightblue">
      <div className="row justify-content-between">
        <div className="col-md-5 mb-2 mb-md-0">
          <h5 className="font-weight-bold secondfont">Become a member</h5>
          Get the latest news right in your inbox. We never spam!
        </div>
        <div className="col-md-7">
          {alert && (
            <div className={`alert alert-${alert.class}`} role="alert">
              {alert.message}
            </div>
          )}

          <div className="row">
            <div className="col-md-12">
              <input
                value={email}
                onChange={(e) => setEmail(e.target.value)}
                type="email"
                className="form-control"
                placeholder="Enter your e-mail address"
                disabled={blocked}
              />
            </div>
            <div className="col-md-12 mt-2">
              <button type="submit" className="btn btn-success btn-block" onClick={subscribe} disabled={blocked}>
                Subscribe
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
