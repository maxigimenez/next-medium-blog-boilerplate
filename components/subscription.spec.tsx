import React from 'react';
import { shallow } from 'enzyme';
import { Subscription } from './subscription';

describe('subscription', () => {
  describe('snapshots', () => {
    let component;
    beforeEach(() => {
      component = shallow(<Subscription />);
    });

    test('render default', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
