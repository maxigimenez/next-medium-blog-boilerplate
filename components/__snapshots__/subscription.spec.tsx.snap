// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subscription snapshots render default 1`] = `
<div
  className="border p-5 bg-lightblue"
>
  <div
    className="row justify-content-between"
  >
    <div
      className="col-md-5 mb-2 mb-md-0"
    >
      <h5
        className="font-weight-bold secondfont"
      >
        Become a member
      </h5>
      Get the latest news right in your inbox. We never spam!
    </div>
    <div
      className="col-md-7"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12"
        >
          <input
            className="form-control"
            disabled={false}
            onChange={[Function]}
            placeholder="Enter your e-mail address"
            type="email"
            value=""
          />
        </div>
        <div
          className="col-md-12 mt-2"
        >
          <button
            className="btn btn-success btn-block"
            disabled={false}
            onClick={[Function]}
            type="submit"
          >
            Subscribe
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
